plugin/audit_log/audit_log.c

```diff
@@ -954,6 +997,36 @@ my_bool audit_log_update_thd_local(MYSQL_THD thd,
     const struct mysql_event_general *event_general=
       (const struct mysql_event_general *) event;
 
+    if (event_general->event_subclass == MYSQL_AUDIT_GENERAL_STATUS)
+    {
+      DBUG_ASSERT(local->stack.frames[local->stack.top].query == NULL
+                  || local->stack.frames[local->stack.top].query
+                      == event_general->general_query.str);
+
+      local->skip_query= audit_log_include_databases
+              && local->stack.frames[local->stack.top].databases_accessed > 0
+              && local->stack.frames[local->stack.top].databases_included == 0;
+
+      local->skip_query|= audit_log_exclude_databases
+              && local->stack.frames[local->stack.top].databases_accessed > 0
+              && local->stack.frames[local->stack.top].databases_excluded
+                 == local->stack.frames[local->stack.top].databases_accessed;
+
+      local->stack.frames[local->stack.top].databases_included= 0;
+      local->stack.frames[local->stack.top].databases_accessed= 0;
+      local->stack.frames[local->stack.top].databases_excluded= 0;
+      local->stack.frames[local->stack.top].query= NULL;
+
+      if (local->stack.top > 0)
+        --local->stack.top;
+
+      if ((event_general->general_command.length == 4 &&
```

Comment:
if (!local->skip_query && (...) to bypass evaluation if we are already skipping the query?